assets/js/atomic/blocks/product/add-to-cart/product-types/variable/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useAddToCartFormContext } from '@woocommerce/base-context';
+
+/**
+ * Internal dependencies
+ */
+import {
+	AddToCartButton,
+	QuantityInput,
+	ProductUnavailable,
+} from '../../shared';
+import VariationPicker from './variation-picker';
+
+/**
+ * Variable Product Add To Cart Form
+ */
+const Variable = () => {
+	const {
+		product,
+		quantity,
+		minQuantity,
+		maxQuantity,
+		setQuantity,
+		formDisabled,
+	} = useAddToCartFormContext();
+
+	if ( product.id && ! product.is_purchasable ) {
+		return <ProductUnavailable />;
+	}
+
+	if ( product.id && ! product.is_in_stock ) {
+		return (
+			<ProductUnavailable
+				reason={ __(
+					'This product is currently out of stock and cannot be purchased.',
+					'woo-gutenberg-products-block'
+				) }
+			/>
+		);
+	}
+
+	return (
+		<>
+			<VariationPicker />
```

Comment:
I think whilst the requirements are not completely clear (some of this may change as we built in extensibility) we should keep them separate and evaluate if they should merge once the picker is merged.